halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java

```diff
@@ -141,6 +158,42 @@ public void saveConfig() {
     saveConfigTo(Paths.get(halconfigPath));
   }
 
+  /**
+   * Deletes all files in the staging directory that are not referenced in the hal config.
+   */
+  public void cleanLocalFiles(Path stagingDirectoryPath) {
+    if (GlobalApplicationOptions.getInstance().isUseRemoteDaemon()) {
```

Comment:
This needs to be configured another way - those config options are used by the CLI - this should use spring.

Reply:
Done.